src/fungible_token.rs

```diff
@@ -426,7 +434,9 @@ impl FungibleToken {
     pub fn accounts_insert(&self, account_id: &str, amount: Balance) {
         if !self.accounts_contains_key(account_id) {
             let key = Self::get_statistic_key();
+            //TODO: verify `read_u64` unwrapping
```

Comment:
I think the way you are unwrapping is fine. When I was writing the `read_u64` type signature I was a little unhappy with the nested `Option<Result<...>>` for exactly this reason. But also that return type corresponds to what is happening (there may or may not be a value, and that value may or may not be a `u64`), so I decided to leave it. If others feel like it is too cumbersome to use then we could switch to a less explicit return type in favour of some error handling inside `read_u64` itself.

Reply:
Another possibility would be to define a type like

```rust
enum MaybeResult<T, E> {
    Some(T),
    Err(E),
    None,
}

impl<T, E> MaybeResult {
    pub fn unwrap_or(self, default: T) -> T {
        match self {
            Self::Some(t) => t,
            Self::Err(_) | Self::None => default,
        }
    }
}
```

But this might be a little overkill if we're only using it once for `read_u64`.

Reply:
Yeah, I don't think we need to define a new type, because in most cases we use just `sdk_unwrap` for it. And there are only a couple of cases where we need to add a couple of sequential `unwrap_or`.

Reply:
Anotherother possibility, and this is probably what I should have done in the first place, would be to capture the possibility of the missing value in the error type.

```rust 
pub(crate) fn read_u64(key: &[u8]) -> Result<u64, ReadU64Error> {
    ...
}

pub(crate) enum ReadU64Error {
    MissingValue,
    InvalidU64,
}
```

Reply:
Yes, probably makes sense to add this as well. But Anyway I suggest doing this in a separate PR to not extend the scope of the current one.

Reply:
https://github.com/aurora-is-near/aurora-engine/pull/118